config/cmake/HDF5Macros.cmake

```diff
@@ -35,6 +35,29 @@ macro (H5_SET_LIB_OPTIONS libtarget libname libtype libpackage)
     endif ()
   endif ()
   HDF_SET_LIB_OPTIONS (${libtarget} ${LIB_OUT_NAME} ${libtype})
+
+  #-- Apple Specific install_name for libraries
+  if (APPLE)
+    option (HDF5_BUILD_WITH_INSTALL_NAME "Build with library install_name set to the installation path" OFF)
+    if (HDF5_BUILD_WITH_INSTALL_NAME)
+      set_target_properties (${libtarget} PROPERTIES
+          INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib"
```

Comment:
Why is this added? The default install name as of CMake 3.0 is `@rpath/lib<name>.dylib` (before it was `lib<name>.dylib`). When users link against the library, it registers `@rpath/lib<name>.dylib` as a load command, so users can set an rpath when linking against HDF5. The situation is then exactly the same as on Linux where users also have to set an rpath.

Reply:
This was added as it was previously removed - can be removed if some apple developer knows this is not needed.